node/core/backing/src/lib.rs

```diff
@@ -0,0 +1,1149 @@
+		let (tx, rx) = oneshot::channel();
+
+		self.tx_from.send(FromJob::RuntimeApiMessage(RuntimeApiMessage::Request(
+				self.parent.clone(),
+				RuntimeApiRequest::SigningContext(tx),
+			)
+		)).await?;
+
+		Ok(rx.await?)
+	}
+
+	/// Request `HeadData` for some `ParaId` from `RuntimeApi`.
+	async fn request_head_data(&mut self, id: ParaId) -> Result<HeadData, Error> {
+		let (tx, rx) = oneshot::channel();
+
+		self.tx_from.send(FromJob::RuntimeApiMessage(RuntimeApiMessage::Request(
+				self.parent.clone(),
+				RuntimeApiRequest::HeadData(id, tx),
+			)
+		)).await?;
+
+		Ok(rx.await?)
+	}
+
+	/// Logic that runs on startup.
+	async fn on_startup(&mut self) -> Result<(), Error> {
+		let validators = self.request_validators().await?;
+	
+		let roster = self.request_validator_groups().await?; 
+
+		self.table_context.key = signing_key(&validators[..], &self.keystore);
+
+		for assignment in roster.scheduled {
+			self.table_context.groups.insert(
+				assignment.para_id,
+				roster.validator_groups[assignment.group_idx.0 as usize]
+				.iter()
+				.map(|idx| validators[*idx as usize].clone())
+				.collect()
+			);
+		}
+
+		if let Some(ref key) = self.table_context.key {
+			for (para_id, group) in self.table_context.groups.iter() {
+				if group.contains(&key.as_ref().public()) {
+					self.assignment = *para_id;
+				}
+			}
+		}
+
+		self.validation_code = self.request_validation_code(self.assignment, 0, None).await?;
+		self.head_data = self.request_head_data(self.assignment).await?;
+		self.table_context.signing_context = self.request_signing_context().await?;
+		self.table_context.validators = validators;
+
+		Ok(())
+	}
+
+	/// Run asynchronously.
+	async fn run(mut self) -> Result<(), Error> {
+		self.on_startup().await?;
+
+		while let Some(msg) = self.rx_to.next().await {
+			match msg {
+				ToJob::CandidateBacking(msg) => {
+					self.process_msg(msg).await?;
+				}
+				_ => break,
+			}
+		}
+
+		Ok(())
+	}
+
+	/// Validate the candidate that is requested to be `Second`ed and distribute validation result.
+	async fn validate_and_second(&mut self, candidate: AbridgedCandidateReceipt) -> Result<(), Error> {
+		let statement = match self.spawn_validation_work(candidate.clone()).await {
+			Ok(()) => {
+				// make PoV available for later distribution. Send data to the availability
+				// store to keep. Sign and dispatch `valid` statement to network if we
+				// have not seconded the given candidate.
+				Statement::Seconded(candidate)
+			}
+
+			// sign and dispatch `invalid` statement to network.
+			Err(_) => Statement::Invalid(candidate.hash()),
+		};
+
+		if let Some(ref signing_key) = self.table_context.key {
+			if let Some(local_index) = self.table_context.local_index() {
+				let signed_statement = SignedFullStatement::sign(
+					statement,
+					&self.table_context.signing_context,
+					local_index,
+					signing_key,
+				);
+
+				self.distribute_signed_statement(signed_statement).await?;
```

Comment:
It doesn't seem to import the statement into the table. Shouldn't it?

Reply:
We might want to track all statements that we've submitted to make sure that we don't submit anything conflicting.